### PR TITLE
Refactor signature

### PR DESCRIPTION
**Pull request by sorhawell**
allow concurrent interface to R have other signatures than fn(Series) -> Series, such as fn(Series, Series) -> Series needed for fold and reduce.

also fix a subtle bug in concurrent R discovered when implementing fold + reduce (cf #403)

### CODE REVIEW

File: src/rust/src/utils/extendr_concurrent.rs [hunk 1]
```diff
@@ -234,7 +236,8 @@ where
                 flume::RecvTimeoutError::Timeout => {
                     //check user interrupts flags in R in a fast high-level way with Sys.sleep(0)
                     let res_res = extendr_api::eval_string("Sys.sleep(0)");
-                    //dbg!(&res_res);
+                    #[cfg(feature = "rpolars_debug_print")]
+                    dbg!(&res_res);
```
Comment (etiennebacher) [inline src/rust/src/utils/extendr_concurrent.rs:240]: should this `dbg!` be commented?

File: src/rust/src/concurrent.rs [hunk 2]
```diff
@@ -11,83 +11,157 @@ use extendr_api::prelude::*;
...
-        extendr_api::error::Error::Other(format!(
-            "provided input is not an R function but a: {:?}",
-            probj.0
-        ))
-    })?;
+// define any possible signature of R lambdas
+#[derive(Debug)]
+pub enum RFnSignature {
+    //... read as function with input 1 Series mapped to output 1 Series
+    FnSeriesTOSeries(ParRObj, pl::Series),
```
Comment (etiennebacher) [inline src/rust/src/concurrent.rs:18]: Is the `TO` on purpose or should it be `FnSeriesToSeries` instead?

**Review by etiennebacher — APPROVED** [review]
I just did some minor formatting in docs, and I have a couple of tiny comments on the Rust side

File: src/rust/src/concurrent.rs (hunk 2 above)
Comment (sorhawell) [reply src/rust/src/concurrent.rs:18]: thx :)

**Pull request merged**